src/core/instance.core.ts

```diff
@@ -60,7 +60,7 @@ export class ZoomPanPinch {
 
   public transformState: ReactZoomPanPinchState;
   public setup: LibrarySetup;
-  public observer: ResizeObserver;
+  public observer!: ResizeObserver;
```

Comment:
I think this is not a good solution for this case since it's not representing true state. We initialize observer with undefined value so in this case it is giving us the falsy assumption that it is always there by `!` casting. You should adjust logic to handle observer being undefined value instead.

Like it was done here: https://github.com/BetterTyped/react-zoom-pan-pinch/pull/485/files#diff-e358387c9a984c08ecacfd998a21a99ffbe7156e0eca96588cedce311d5a65acL160

Reply:
Because the observer already exists after initialization.
The code here does not have conditional branches like before: https://github.com/nuintun/react-zoom-pan-pinch/blob/0a2e348e1847e6bb31283104fd0b88fe1fdea796/src/core/instance.core.ts#L199

Reply:
The problem is that initialization is a method, not a constructor so it could be triggered at different times - for example, just after rendering.
So the lifecycle is like:
1. We make instance `new ReactZoomPanPinch()`
2. Reacts lifecycle kicks in and renders the content for x time
3. Then we trigger initialize on useLayoutEffect - this is the time when wrapper and content is mounted to DOM

Between 1 and 3 there is some processing on the React's side which is taking unspecified amount of time based on the component you're building. During this time `observer` is undefined so this implementation is not truthy. 

So my approach would be to keep it as close as possible to the real flow. Typescript must reflect the real world otherwise it is becoming a problem to handle later on.

Reply:
OK, I will fixed this.